### DISCUSSION TITLE
How do you export as .pdf?

### DISCUSSION THREAD

**Discussion by ZebulaCodes** [EvotecIT/PSWriteHTML#304] [Q&A]
I can show the report as HTML but I'd like to convert it automatically to .pdf without having to manually click the .pdf button on the HTML report.

**Comment by PrzemyslawKlys** [comment]
There's no way to do this in PSWriteHTML. You can generate PDF using https://github.com/EvotecIT/PSWritePDF

**Reply by ZebulaCodes** [reply]
I just took a quick look through PSWritePDF, I see many ways to create one from scratch but is there a way to convert an existing file, say an HTML report, into pdf? Maybe I haven't found that part of the example list yet.

**Reply by PrzemyslawKlys** [reply]
No. But from what I can see there's an iText addon that could do that: https://kb.itextpdf.com/home/it7kb/ebooks/itext-7-converting-html-to-pdf-with-pdfhtml 

I've opened an issue: https://github.com/EvotecIT/PSWritePDF/issues/32 

But if you want it to look good, you should recreate the logic in PDF directly. The PDF button in PSWriteHTML basically converts Table to PDF, which in PSWritePDF is 3 lines of code. Not sure why would you need to go thru conversion process (which as any conversion process brings many risks, and assumptions) instead of using 3 lines of code (to add a table + some lines to make some introduction). 

```powershell
New-PDF {
    New-PDFText -Text 'Hello ', 'World' -Font HELVETICA, TIMES_ITALIC -FontColor GRAY, BLUE -FontBold $true, $false, $true
    New-PDFText -Text 'Testing adding text. ', 'Keep in mind that this works like array.' -Font HELVETICA -FontColor RED
    New-PDFText -Text 'This text is going by defaults.', ' This will continue...', ' and we can continue working like that.'

    New-PDFText -Text 'This table is representation of ', 'PSCustomObject', ' or other', ', ', 'standard types' -FontColor BLACK, RED, BLACK, BLACK, RED -FontBold $false, $true, $false, $false, $true

    New-PDFTable -DataTable $DataTable123

    New-PDFText -Text 'This shows how to create a list' -FontColor MAGENTA

    New-PDFList -Indent 3 {
        New-PDFListItem -Text 'Test'
        New-PDFListItem -Text '2nd'
    }

    New-PDFText -Text 'This table is representation of ', 'Array of Hashtable/OrderedDictionary' -FontColor BLACK, BLUE

    New-PDFTable -DataTable $DataTable1234

    New-PDFText -Text 'This table is representation of ', 'Hashtable/OrderedDictionary' -FontColor BLACK, BLUE

    New-PDFTable -DataTable $DataTable12345

} -FilePath "$PSScriptRoot\Example06.pdf" -Show
```

**Comment by PrzemyslawKlys** [comment]
I've started playing with this idea and it seems to work. Should be in next version, but I would advice hand-drawn stuff using New-PDF
